system/src/commonMain/kotlin/com/splendo/kaluga/system/network/NetworkState.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network
+
+import com.splendo.kaluga.state.HandleAfterNewStateIsSet
+import com.splendo.kaluga.state.HandleBeforeOldStateIsRemoved
+import com.splendo.kaluga.state.State
+
+sealed class NetworkState(open val networkType: Network, private val networkManager: BaseNetworkManager) : State() {
```

Comment:
I think we are missing an "Initializing/Unknown" state here, but correct me if I'm wrong, do we know the state immediately? I see a call to `networkManager.isNetworkEnabled()` in init, but it seems to me this will just contain some default.

Reply:
No, the initial state would be `Unavailable` because `lastKnownNetwork` initialValue is Network.Absent. I agree that this should be handled with an Unknown state

Reply:
Not sure what you are saying "no" to when it's exactly as I say?

Reply:
You asked me a question, I replied explaining how it works at the moment and agreed with your point.

> do we know the state immediately?

Reply:
Ah like that 👍

Reply:
eb20ec4 f217621 7d09878 b72497d